shapely/errors.py

```diff
@@ -31,3 +31,7 @@ class TopologicalError(ShapelyError):
 
 class PredicateError(ShapelyError):
     """A geometric predicate has failed to return True/False."""
+
+
+class GeometryAlreadyPreparedError(ShapelyError):
+    """Attempted to prepare an already prepared geometry"""
```

Comment:
@snorfalorpagus how about a more generic GeometryPreparationError ?

Reply:
Thinking about this a little more, I think we could just return the object without an error if it is an instance of PreparedGeometry, analogous to the result of `int(4)` or `float(3.14)`.